File: CustomisedCollections/src/dsa/MyQueue.java
package dsa;

import java.util.NoSuchElementException;

public class MyQueue {
    private int count;
    private String[] elements;

    public MyQueue(int maxSize){
        elements = new String[maxSize];
    }
    public boolean add(String element) {
        if(count != elements.length) {
            elements[count++] = element;
            return true;
        }
        else throw new IllegalStateException("Queue is full");
    }

    public boolean offer(String element) {
        if(count != elements.length) {
            elements[count++] = element;
            return true;
        }
        else return false;
    }

    public String remove() {
        if(count != 0) {
            String firstElement = elements[0];
            for(int counter = 1; counter < count; counter++) {
                elements[counter - 1] = elements[counter];
            }
            count --;
            return firstElement;
        }
        else throw new NoSuchElementException("Queue is empty");
    }

    public String poll() {
        if(count != 0){
            String firstElement = elements[0];
            for(int counter = 1; counter < count; counter++) {
                elements[counter - 1] = elements[counter];
            }
            count --;
            return firstElement;
        }
        else return null;
    }

    public String element() {
        if(count != 0) return elements[0];
        else throw new NoSuchElementException("Queue is empty");
    }

    public String peek() {
        if(count != 0) return elements[0];
        else return null;
    }
}

File: CustomisedCollections/src/dsa/my_queue.py
class MyQueue:
    def __init__(self, max_size):
        self.queue = []
        self.count = 0
        self.max_size = max_size

    def add(self, element):
        if self.count != self.max_size:
            self.queue.append(element)
            self.count += 1
            return True
        else:
            raise OverflowError("Queue is full")

    def offer(self, element):
        if self.count != self.max_size:
            self.queue.append(element)
            self.count += 1
            return True
        else:
            return False

    def remove(self):
        if self.count != 0:
            element = self.queue[0]
            for count in range(1, self.count):
                self.queue[count-1] = self.queue[count]
            self.count -= 1
            return element
        else:
            raise OverflowError("Queue is empty")

    def poll(self):
        if self.count != 0:
            element = self.queue[0]
            for count in range(1, self.count):
                self.queue[count-1] = self.queue[count]
            self.count -= 1
            return element
        else:
            return None

    def element(self):
        if self.count != 0:
            return self.queue[0]
        else:
            raise OverflowError("Queue is empty")

    def peek(self):
        if self.count != 0:
            return self.queue[0]
        else:
            return None